packages/devtools_app/test/chart_test.dart

```diff
@@ -0,0 +1,931 @@
+                    ),
+                  ),
+                ],
+              );
+            })));
+
+        await tester.pumpAndSettle();
+      }
+
+      void chartAllData(ChartController controller) {
+        for (var datumIndex = 0;
+            datumIndex < memoryJson.data.length;
+            datumIndex++) {
+          final datum = memoryJson.data[datumIndex];
+
+          final external = datum.external.toDouble();
+          final used = datum.used.toDouble();
+          final capacity = datum.capacity.toDouble();
+          final rss = datum.rss.toDouble();
+
+          controller.timestamps.add(datum.timestamp);
+
+          _rawExternal.add(Data(datum.timestamp, external));
+
+          addDataToTrace(controller, externalTraceIndex, _rawExternal.last);
+
+          _rawUsed.add(Data(datum.timestamp, external + used));
+          addDataToTrace(controller, usedTraceIndex, _rawUsed.last);
+
+          _rawCapacity.add(Data(datum.timestamp, capacity));
+          addDataToTrace(controller, capacityTraceIndex, _rawCapacity.last);
+
+          _rawRSS.add(Data(datum.timestamp, rss));
+          addDataToTrace(controller, rssTraceIndex, _rawRSS.last);
+        }
+      }
+
+      Future<void> setupScaledChart(
+          WidgetTester tester, ChartController controller, Key chartKey) async {
+        final theChart = Chart(controller, 'Scaled Chart');
+
+        setupTraces(controller);
+
+        loadData();
+
+        await pumpChart(tester, chartKey, theChart, 250);
+
+        expect(find.byWidget(theChart), findsOneWidget);
+
+        // Validate the X axis before data added.
+        expect(controller.visibleTicks, equals(214));
+        expect(controller.xCanvasChart, equals(50.0));
+        expect(controller.xPaddingRight, equals(0.0));
+        expect(controller.displayXLabels, true);
+        expect(controller.canvasChartWidth, equals(2140.0));
+
+        // Validate the Y axis before data added.
+        expect(controller.yScale.computedMin, equals(0.0));
+        expect(controller.yScale.computedMax, equals(0.0));
+        expect(controller.yScale.labelTicks, equals(0.0));
+        expect(controller.yScale.labelUnitExponent, 0.0);
+        expect(controller.yScale.tickSpacing, equals(1.0));
+        expect(controller.yScale.maxPoint, equals(0.0));
+        expect(controller.yScale.maxTicks, equals(165.0));
+
+        chartAllData(controller);
+      }
+
+      /// Validate the labels displayed on the y-axis.
+      void validateScaledYLabels(ChartController controller) {
+        // Validate the labels displayed on the y-axis.
+        final yScale = controller.yScale;
+        expect(yScale.labelTicks, equals(7));
+        for (var labelIndex = yScale.labelTicks;
+            labelIndex >= 0;
+            labelIndex--) {
+          final labelName = ChartPainter.constructLabel(
+            labelIndex.toInt(),
+            yScale.labelUnitExponent.toInt(),
+          );
+
+          switch (labelIndex.toInt()) {
```

Comment:
nit: instead of writing a switch write
expectedLabels = ['0', '100M',...];
expect(labelName, expectedLabel[labelIndex.toInt()]);

Reply:
Done.